### ISSUE TITLE
False negative: In a stub file `@final` must be applied only to the first overload

### ISSUE THREAD

**Issue by hamdanal** [microsoft/pyright#6866]
**Describe the bug**
pyright does not generate an error if `@final` appears on subsequent overloads. [PEP 591](https://peps.python.org/pep-0591/#id1) says:
*"For overloaded methods, `@final` should be placed on the implementation (or on the first overload, for stubs)"*

**Code or Screenshots**
Consider this pyi file:

```python
from typing import final, overload

class C:
    @overload
    def f(self, x: int) -> int: ...
    @final
    @overload
    def f(self, x: str) -> str: ...

```

```console
$ pyright t.pyi
0 errors, 0 warnings, 0 informations
$ mypy t.pyi
t.pyi:6: error: In a stub file @final must be applied only to the first overload  [misc]
```

**VS Code extension or command-line**
command-line: pyright 1.1.344

**Comment by erictraut** [comment]
The spec doesn't disallow `@final` from being applied to subsequent overloads, so I don't think that should be flagged as an error. However, I agree that pyright should report an error for cases where the first overload doesn't have a `@final` decorator but subsequent ones do.

I will also add this condition to the [typing spec conformance tests](https://github.com/python/typing/pull/1554) that I've been working on.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
